### PR TITLE
Standardization of CI and logging

### PR DESCRIPTION
**Pull request by teutoburg**
- Use new DevOps split tests and updated dependency tests.
- Use logger factory from astar-utils

### CODE REVIEW

**Review by hugobuddel — APPROVED** [review]
Good!

I'm a bit questioning the use of prereleases for astar-utils though. I think it would be better to just release astar-utils often and not allow prereleases in the other projects. Because I think we should not have prerelease dependencies in our releases, which would currently requiring updating pyproject.toml before we can make a release. This breaks the automated release plan we are working towards.

Or vice-versa, I don't see the benefit of making pre-releases of astar-utils, but I do see drawbacks. I'm probably missing something.

**Comment by teutoburg** [comment]
> I'm a bit questioning the use of prereleases for astar-utils though. I think it would be better to just release astar-utils often and not allow prereleases in the other projects. Because I think we should not have prerelease dependencies in our releases, which would currently requiring updating pyproject.toml before we can make a release. This breaks the automated release plan we are working towards.

My original thought with the prereleases here was twofold: Firstly I wasn't sure if I wanted to include more in v0.2.0 of astar-utils, but now I don't think that's the case. Thinking about this again, it probably wasn't a very useful idea, because of the issues you mentioned. Would have been fine to simply release astar-utils v0.2.0 and then make v0.2.1 or v0.3.0 or whatever is appropriate. Especially now that the auto-PyPI stuff seems to work properly. Secondly, I wanted to test the complete ensemble to make sure things work together. But that would have been fine in a local branch tbh.

So in summary, I'm not planning to make any more pre-release-dependencies unless I find a good reason to do so 🙃

**Comment by hugobuddel** [comment]
I get the desire to test the whole ensemble before 'committing' to a true release.

One idea behind ScopeSim_Data is to do just that: install all our packages from the default branch and run all tests and notebooks. The primary goal is to subsequently store the data, but it also doubles as a full test suite, that is also easy to run locally.

However, astar-utils is not explicitly installed by ScopeSim data, so it is simply installed as an dependency from PyPI. Perhaps we should add astar-utils to `download_scopesim_data.sh`, then you'd get a nightly full ensemble test run for free!
